File: web/HIGN/HIGN/Personnel/Application/Internal/CommandServices/ExaminerCommandService.cs
using HIGN.Personnel.Domain.Model.Aggregates;
using HIGN.Personnel.Domain.Model.Commands;
using HIGN.Personnel.Domain.Model.Exceptions;
using HIGN.Personnel.Domain.Repositories;
using HIGN.Personnel.Domain.Services;
using HIGN.Shared.Domain.Repositories;

namespace HIGN.Personnel.Application.Internal.CommandServices;

public class ExaminerCommandService(IUnitOfWork unitOfWork, IExaminerRepository examinerRepository) : IExaminerCommandService
{
    public async Task<Examiner?> Handle(CreateExaminerCommand command)
    {
        if (examinerRepository.ExistsByNationalProviderIdentifier(command.NationalProviderIdentifier))
        {
            throw new ExaminerWithThisNationalProviderIdentifierAlreadyExistsException();
        }
        var examiner = new Examiner(command);
        await examinerRepository.AddAsync(examiner);
        await unitOfWork.CompleteAsync();
        return examiner;
    }
}

File: openfinalfinal/u202120418/u202120418/src/main/java/org/hign/platform/u202120418/personnel/domain/services/ExaminerCommandService.java
package org.hign.platform.u202120418.personnel.domain.services;

import org.hign.platform.u202120418.personnel.domain.model.aggregates.Examiner;
import org.hign.platform.u202120418.personnel.domain.model.commands.CreateExaminerCommand;

import java.util.Optional;

/**
 * Commands service for examiner
 * @author Fiorella Jarama Peñaloza
 * @version 1.0
 */
public interface ExaminerCommandService {
    Optional<Examiner> handle(CreateExaminerCommand command);
}
